src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -987,7 +978,8 @@ ucs_status_t uct_dc_mlx5_ep_pending_add(uct_ep_h tl_ep, uct_pending_req_t *r,
                 return UCS_ERR_BUSY;
             }
         } else {
-            if (uct_dc_mlx5_iface_dci_ep_can_send(ep)) {
+            if (uct_dc_mlx5_iface_dci_ep_can_send(ep) &&
+                (iface->super.super.tx.ops_available > 0)) {
```

Comment:
this is a bit suspicious, it does not exactly match the condition for sending and processing pending operations

Reply:
Yes, but we should be able to add flush to pending when no tx ops available. Without  this change, pending add returns BUSY, while flush itself returns NO_RESOURCES

Reply:
so let's add the check of ops_available to uct_rc_iface_has_tx_resources()?

Reply:
I'd not block any operation (except flush) due to lack of tx ops. Otherwise flushes would have severe impact on performance.

Reply:
it's only used for processing pending operations, not for plain send operations

Reply:
right, missed that